assets/events.py

```diff
@@ -0,0 +1,203 @@
+        self.command_name = command_name
+        self.command_argument = command_argument
+        # Namespace the approval lock
+        self.data = json.loads(json_data)
+        self.token = None
+
+        self.api_url = None
+        self.api_login = None
+        self.api_password = None
+        self.organization = None
+        self.icon = None
+        self.messagel = None
+        self.severity = None
+        self.tags = None
+        self.title = None
+        self.type = None
+
+        # allow debug logging to console for tests
+        if os.getenv('RESOURCE_DEBUG', False) or self.data.get('source', {}).get('debug', False):
+            log.basicConfig(level=log.DEBUG)
+        stderr = log.StreamHandler()
+        stderr.setLevel(log.INFO)
+        log.getLogger().addHandler(stderr)
+
+        log.debug('command: %s', command_name)
+        log.debug('input: %s', self.data)
+        log.debug('args: %s', command_argument)
+        log.debug('environment: %s', os.environ)
+
+
+    def check_cmd(self, source, version):
+        """
+        Check for new version(s)
+
+        #TODO this first release of the resource won't check for new events/won't trigger
+        But in a future version, we may want to check/filter specific events to trigger the pipeline
+
+        :param source: is an arbitrary JSON object which specifies the location of the resource,
+        including any credentials. This is passed verbatim from the pipeline configuration.
+        :param version: is a JSON object with string fields, used to uniquely identify an instance of the resource.
+        :return: a dict with the version fetched
+        """
+
+        log.debug('version: %s', version)
+
+        if not version:
+            version = {"timestamp": '0'}
+        log.debug('source: %s', source)
+        log.debug('version: %s', version)
+
+        return versions_list
+
+
+    def in_cmd(self, target_dir, source, version, params):
+        """
+        Get an event from Cycloid API
+
+        #TODO this first release of the resource won't get an event, we don't have usecase right now
+
+        :param target_dir: a temporary directory which will be exposed as an output
+        :param source: is the same value as passed to check
+        :param version: is the same type of value passed to check, and specifies the version to fetch.
+        :param params: is an arbitrary JSON object passed along verbatim from params on a get.
+        :return: a dict with the version fetched and the metadata
+        """
+        log.debug('source: %s', source)
+        log.debug('version: %s', version)
+
+        if not version:
+            version = {'timestamp': '0'}
+
+        metadata = []
+
+        metadata_path = os.path.join(target_dir, 'metadata')
+        with open(metadata_path, 'w') as metadata_file:
+            json.dump(metadata, metadata_file)
+
+        return {
+            'version': {'timestamp': '0'},
```

Comment:
shouldn't this be 
```python
return {
    'version': version,
    'metadata': metadata,
}
```

?

Reply:
no, out does not get version as param from concourse.

Put should produce and return a uniq version number which will be used by the get and check function.

But the current implementation of events on CW backend does not provide such uniq ID to get a specific event.

We could discuss it in #2